### ISSUE TITLE
Item "duplication" with graves

### ISSUE THREAD

**Issue by staviq** [NillerMedDild/Enigmatica6#3694]
### Modpack Version

0.5.20

### Describe your issue.

If after dying, player switches to spectator mode and back, soulbound items will get weird when clicking on the grave.

Example: I had two swords, one diamond i bought from traveling merchant, and one i looted, both diamond enchanted swords.
When i died, i did F3+N to get to my items, and when i switched back from F3N, clicking my grave replaced one of my diamond swords with the other and now i have two identical enchanted diamond swords instead of two diffrent enchanted diamond swords. Essentially one item turned into the other one, but the original item was kept.

"Duplicated" items persist through server restart.

I restored world from backup and tested it again and the problem is reproducible.

### Have you modified the modpack?

No

### Did the issue happen in singleplayer or on a server?

Server

**Comment by MuteTiefling** [comment]
Thanks for the report. Are you certain that spectator mode is a requirement for this? You weren't in it when you claimed the items, if I'm understanding properly.

**Comment by staviq** [comment]
> Thanks for the report. Are you certain that spectator mode is a requirement for this? You weren't in it when you claimed the items, if I'm understanding properly.

When i realized i have two identical swords in my inventory, i rolled back my world from backup several times until i found one backup that had my two swords intact, then i went and died on purpose, i then respawned in my bed, went on foot to where i died and picked up my grave, and everything was fine. So then i again got myself killed, this time i went into spectator mode to get out of my house quicker, then outside the house i switched back from f3n to normal mode and on foot approached my grave and picked up my items, and then the swords got mixed up.

So i would seem like switching to spectator mode and back is requirement for reproducing this problem, but i haven't tested it enough to confirm that this is the only way to cause this problem.

**Comment by MuteTiefling** [comment]
I'm having a hard time reproducing this. 

- Fresh world
- Give myself two diamond swords. One enchanted with sharpness, the other enchanted with unbreaking.
- Set spawn with bed
- Go die

Upon dying, I respawn and collect my things without fault even if I enter spectator mode and leave it prior to collecting my items.

**Comment by Darkere** [comment]
Soulbound is probably also a requirement for this bug.

**Comment by MuteTiefling** [comment]
Oh good call. Will try that

**Comment by MuteTiefling** [comment]
Yeah, even with soulbound on the weapons I'm getting back exactly the tools I had at death. 

Also tried clicking the grave vs sneaking on it. No difference. I get the right stuff back... Still, will report to Quarris, just might be harder to track down if we can't reproduce.
